src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBasicDemo.java

```diff
@@ -0,0 +1,172 @@
+import static io.cdap.plugin.utils.GCConstants.TABLE_DEL_MSG;
+
+/**
+ * DemoTestCase GCP.
+ */
+public class GCSBasicDemo implements CdfHelper {
+  GcpClient gcpClient = new GcpClient();
+
+  @Given("Open Datafusion Project to configure pipeline")
+  public void openDatafusionProjectToConfigurePipeline() throws IOException, InterruptedException {
+    openCdf();
+  }
+
+  @When("Source is GCS bucket")
+  public void sourceIsGCSBucket() throws InterruptedException {
+    CdfStudioActions.selectGCS();
+  }
+
+  @When("Target is BigQuery")
+  public void targetIsBigQuery() {
+    CdfStudioActions.sinkBigQuery();
+  }
+
+  @Then("Link Source and Sink to establish connection")
+  public void linkSourceAndSinkToEstablishConnection() {
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.toBigQiery);
+    SeleniumHelper.dragAndDrop(CdfStudioLocators.fromGCS, CdfStudioLocators.toBigQiery);
+  }
+
+  @Then("Enter the GCS Properties with {string} GCS bucket and {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAnd(String bucket, String formatType) throws
+    IOException, InterruptedException {
+    CdfGcsActions.gcsProperties();
+    CdfGcsActions.enterReferenceName();
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.getGcsBucket(CdapUtils.pluginProp(bucket));
+    CdfGcsActions.selectFormat(CdapUtils.pluginProp(formatType));
+    CdfGcsActions.skipHeader();
+    CdfGcsActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(CdfGCSLocators.getSchemaButton);
+  }
+
+  @Then("Run and Preview")
+  public void runAndPreview() {
+    CdfStudioActions.runAndPreviewData();
+  }
+
+  @Then("Save and Deploy Pipeline")
+  public void saveAndDeployPipeline() {
+    CdfStudioActions.pipelineName();
+    CdfStudioActions.pipelineNameIp("TestPipeline" + UUID.randomUUID());
+    CdfStudioActions.pipelineSave();
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Run the Pipeline in Runtime")
+  public void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till pipeline is in running state")
+  public void waitTillPipelineIsInRunningState() {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 1000000);
+    wait.until(ExpectedConditions.or(ExpectedConditions.visibilityOfElementLocated(
+      By.xpath("//*[@data-cy='Succeeded']")), ExpectedConditions.visibilityOfElementLocated(
+      By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Verify the pipeline status is {string}")
+  public void verifyThePipelineStatusIs(String status) {
+    boolean webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("Open Logs")
+  public void openLogs() throws FileNotFoundException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+    PrintWriter out = null;
+    try {
+      out = new PrintWriter(BeforeActions.myObj);
```

Comment:
rename myObj -> something more descriptive. also use a getter

Reply:
relevant changes made in https://github.com/cdapio/cdap-e2e-tests/pull/51 and https://github.com/data-integrations/google-cloud/pull/861